public/app/core/components/AppChrome/ReturnToPrevious/DismissableButton.tsx

```diff
@@ -0,0 +1,58 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Button, ButtonGroup, useStyles2 } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+
+export interface DismissableButtonProps {
+  label: string;
+  onClick: () => void;
+  onDismiss: () => void;
+}
+
+export const DismissableButton = ({ label, onClick, onDismiss }: DismissableButtonProps) => {
+  const styles = useStyles2(getStyles);
+
+  return (
+    <ButtonGroup className={styles.buttonGroup}>
+      <Button
+        icon="angle-left"
+        size="sm"
+        variant="primary"
+        fill="outline"
+        onClick={onClick}
+        title={label}
+        className={styles.mainDismissableButton}
+      >
+        {label}
+      </Button>
+      <Button
+        icon="times"
+        aria-label={t('dismissable-button.close', 'Close')}
+        variant="primary"
+        fill="outline"
+        size="sm"
+        onClick={onDismiss}
+      />
+    </ButtonGroup>
+  );
+};
+const getStyles = (theme: GrafanaTheme2) => ({
+  mainDismissableButton: css({
+    width: '100%',
+    ['span']: {
```

Comment:
is it possible to use `<Text truncate>` here? at the very least we should make this `> span` to only target the immediate span child instead of every `span` inside the button

Reply:
The `truncate` prop of the `Text` component does not work if the element is a `span`. I'll add the `>` to just affect the first `span` child, thanks!